### ISSUE TITLE
AInspector no longer highlights WCAG violations

### ISSUE THREAD

**Issue by mcchipper** [opena11y/ainspector-for-firefox#54]
Our organization is currently using the latest version of AInspector 2.99. When selecting to view the Details and then selecting V/W, the areas/elements in violation used to be highlighted. This was very helpful for QA (we perform manual testing) and Development in correcting these accessibility issues.

The previous versions had the highlighting and would like to have this feature restored.

If the feature is still available and we're just not seeing it, could you provide troubleshooting tips to get it to work.

Thank You

Chip McConnell
Quality Assurance Analyst CollegeSource, Inc.
chip@collegesource.com

**Comment by jongund** [comment]
@mcchipper 
Thank you for your feedbac I will look into this issue.

**Comment by jongund** [comment]
@mcchipper 
I went to the following URL:
https://collegesource.com/

I am viewing the Landmark Category Rule: All content must be in a landmark and turned on the V/W option and it is highlighting the Violating elements on the page.

Are you able to replicate what I am seeing?

**Comment by mcchipper** [comment]
@jongund 

Our application wasn't showing any Landmarks violation so I went to the Amazon page and did see the highlights for that category. However, it appears the other categories I tested (Colors/Content, Tables/Layout, ARIA Widgets and Site Navigation) do not show highlights.

**Comment by jongund** [comment]
@mcchipper 
Is there a link you can send me so I can test the pages?

**Comment by mcchipper** [comment]
@jongund 

Unfortunately, there's no way I can give you access to our environments. However, we have an application that is free to use. You will need to create a user id and password. The application is for high school and college students to see if there courses transfer to other schools or if they can find replacement courses.
The application is moderately intuitive and there are videos on the index page that walk you through how to use it.
The URL is:
https://www.transferology.com

I apologize for the hoop-jumping for troubleshooting this, but it's the only way I can give you access to our application.

If you need anything else, please let me know.

Thank you

**Comment by jongund** [comment]
@mcchipper 

I can see the problem now, thank you for reporting it.  I am working on a solution.

**Comment by mcchipper** [comment]
@jongund 

You rock dude!

**Comment by jongund** [comment]
@mcchipper 

I think I fixed the problem, it is working for me on the URL you provided.
It will be version 2.99.1, it should be available soon from the add-ons website.
After you have it installed, please test it and let me know if you still have the problem.
Thanks. again for your help!

**Comment by mcchipper** [comment]
Thank you! I will let you know how my testing goes.

**Comment by mcchipper** [comment]
@jongund

I just wanted to let you know it appears the highlighting is working again. Thank you for your quick response and fix for this issue. It is greatly appreciated.

I have been reviewing the Preferences>Ruleset Options and have a question.
I've been trying to find documentation on the differences in the Rule Scope filter selections. The All rules selection seems to identify the most violations as compared to Page scope and Website scope. I don't know what that means in terms of what the previous version of AInspector used to provide. 

If you have documentation on the specifics of these settings, could you please send me a link. AInspector is my go to tool for accessibility because of it's ease of use and the fact that it highlights all of the elements associated within a category violation. It's very useful for me and the developers use it extensively as well.

Thank you again for your help.

**Comment by jongund** [comment]
@mcchipper 
Thank you for reporting this issue.
It may have to do with changes in the CSS positioning property.
Is there a  public URL where I can replicate what you are seeing?

**Comment by mcchipper** [comment]
@jongund 

I'm sorry, but unfortunately the screen cap of the application I gave you is not accessible without a paid license and I have no way of allowing you access to look at it.
What I did notice is that when the Forms category (where the violation is) is first initiated, the highlight is on the icon:
![image](https://github.com/opena11y/ainspector-for-firefox/assets/146010593/1e73b9c2-c03a-4f36-83b3-558fa64f933b)

But when I select to view the next violation, the highlight is not on the next offending element:

![image](https://github.com/opena11y/ainspector-for-firefox/assets/146010593/69d234cf-e455-4b18-b9a1-c98e51cbb8ac)

And if I return to the initial violation, the highlight is off again:
![image](https://github.com/opena11y/ainspector-for-firefox/assets/146010593/ccfb592e-17b6-4f6d-bdb5-0efdc5d857be)

Again, I apologize for making this troubleshooting more difficult, but because of the limitations on who may access our applications there isn't much I can do.

Any help you can give is greatly appreciated.

Thanks

@mcchipper

**Comment by jongund** [comment]
@mcchipper 
I am moving this to a new issue.

**Comment by jongund** [comment]
Issue has been fixed in [AInspector 2.99.1](https://addons.mozilla.org/en-US/firefox/addon/ainspector-wcag/).
